### PR TITLE
fix(TDI-48610):remove some optional transitive dependency from daikon bundle

### PR DESCRIPTION
**Pull request by jzhao-talend**
**What is the problem this Pull Request is trying to solve?**

-  remove some optional transitive dependency from daikon bundle

**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDI-48610
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Review by wwang-talend — DISMISSED** [review]

**Comment by sgandon** [comment]
Hello,
Can you explain more the reason for this please ?
In you description you are 
saying 

> remove some optional transitive dependency from daikon bundle
as the problem you are trying to soolve but this is not the problem but more the chosen solution.
Can you elaborate on the reason you are doing this please ? cause I am afraid removing imports will make some API fail to work in the bundle.

**Comment by jzhao-talend** [comment]
> Hello, Can you explain more the reason for this please ? In you description you are saying
> 
> > remove some optional transitive dependency from daikon bundle
> 
> as the problem you are trying to soolve but this is not the problem but more the chosen solution. Can you elaborate on the reason you are doing this please ? cause I am afraid removing imports will make some API fail to work in the bundle.

tried to update daikon directly:
https://jenkins-connectors.datapwn.com/job/components/job/build/job/jzhao%252FTDI-48610_crypto-utils_6.8.11/17/console
osgi test failed which was caused by commons-compress optional dependencies
tried to include them in the test. but seems:
zstd-jni **Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=11.0))**
and our tcompv0 projects build with JDK 8

**Comment by sgandon** [comment]
I am afraid that the import that you remove will be required for some API use by other teams so it is quite dangerous to remove those just because you test case is failing.
Can you just remove the optional bundle `zstd-jni` from your test suite instead of changing daikons bundle imports ?

**Comment by jzhao-talend** [comment]
> I am afraid that the import that you remove will be required for some API use by other teams so it is quite dangerous to remove those just because you test case is failing. Can you just remove the optional bundle `zstd-jni` from your test suite instead of changing daikons bundle imports ?

do you have any guidance to resolve this error?
```
ERROR: Bundle org.talend.daikon [32] Error starting file:/tmp/1665307932646-0/pax-exam-downloads/org.talend.daikon_6.8.11.jar (org.osgi.framework.BundleException: Unable to resolve org.talend.daikon [32](R 32.0): missing requirement [org.talend.daikon [32](R 32.0)] osgi.wiring.package; (osgi.wiring.package=com.github.luben.zstd) Unresolved requirements: [[org.talend.daikon [32](R 32.0)] osgi.wiring.package; (osgi.wiring.package=com.github.luben.zstd)])
```

**Comment by sgandon** [comment]
I would suggest that you make those import optional just like you mention in the title of ticket instead of removing them.

File: daikon/pom.xml [hunk 1]
```diff
@@ -233,7 +233,7 @@
                                 <Embed-Dependency>*;artifactId=paranamer|snappy-java|commons-compress|xz;inline=true
                                 </Embed-Dependency>
                                 <Import-Package>
-                                    com.cedarsoftware.*;resolution:=optional,com.fasterxml.*;resolution:=optional,org.apache.*;resolution:=optional,*
+                                    !org.objectweb.asm;!com.github.luben.zstd.*;!org.tukaani.xz.*;!org.brotli.*;com.cedarsoftware.*;resolution:=optional,com.fasterxml.*;resolution:=optional,org.apache.*;resolution:=optional,*
```
Comment (sgandon) [inline daikon/pom.xml:236]:
```suggestion
                                    org.objectweb.asm;resolution:=optional,com.github.luben.zstd.*;resolution:=optional,org.tukaani.xz.*;resolution:=optional,org.brotli.*;resolution:=optional,com.cedarsoftware.*;resolution:=optional,com.fasterxml.*;resolution:=optional,org.apache.*;resolution:=optional,*
```
I would suggest you make those import optional

**Review by sgandon — APPROVED** [review]
LGTM

**Pull request merged**